godot-core/src/builtin/dictionary.rs

```diff
@@ -0,0 +1,360 @@
+use godot_ffi as sys;
+
+use crate::builtin::{inner, FromVariant, ToVariant, Variant, VariantConversionError};
+use crate::obj::Share;
+use std::collections::{HashMap, HashSet};
+use std::fmt;
+use sys::types::*;
+use sys::{ffi_methods, interface_fn, GodotFfi};
+
+use super::Array;
+
+/// Godot's `Dictionary` type.
+///
+/// The keys and values of the array are all `Variant`, so they can all be of different types.
+///
+/// Keys are assumed to be cheap to clone, this will usually be the case especially for
+/// value types and reference counted types.
```

Comment:
Why exactly keys? Since both keys and values are variants, there is no difference.

> this will usually be the case especially for value types and reference counted types.

I would not mention this. This is explained more precisely in the lib.rs documentation (#93).

We could just say:
```rs
/// The keys and values of the array are all `Variant`, so they can all be of different types.
/// Variants are designed to be generally cheap to clone.
```

Reply:
Mainly cause i clone keys in various places, but i don't clone values ever. However that's fair.